Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -333,22 +407,30 @@ class FakespotViewController:
 
         case .qualityDeterminationCard:
             let reviewQualityCardView: FakespotReviewQualityCardView = .build()
+            viewModel.reviewQualityCardViewModel.expandState = state.isReviewQualityExpanded ? .expanded : .collapsed
             viewModel.reviewQualityCardViewModel.dismissViewController = { [weak self] in
                 guard let self = self else { return }
                 self.delegate?.fakespotControllerDidDismiss(animated: true)
             }
+            viewModel.reviewQualityCardViewModel.onExpandStateChanged = { state in
+                store.dispatch(FakespotState.Action.reviewQualityDidChange)
+            }
             reviewQualityCardView.configure(viewModel.reviewQualityCardViewModel)
             return reviewQualityCardView
 
         case .settingsCard:
             let view: FakespotSettingsCardView = .build()
-            view.configure(viewModel.settingsCardViewModel)
+            viewModel.settingsCardViewModel.expandState = state.isSettingsExpanded ? .expanded : .collapsed
             viewModel.settingsCardViewModel.dismissViewController = { [weak self] action in
                 guard let self = self else { return }
                 self.delegate?.fakespotControllerDidDismiss(animated: true)
                 guard let action else { return }
                 viewModel.recordDismissTelemetry(by: action)
             }
+            viewModel.settingsCardViewModel.onExpandStateChanged = { state in
+                store.dispatch(FakespotState.Action.settingsStateDidChange)
```

Comment:
This was a bit strange to see. Having the global `store` in the controller makes it impossible for features to be extracted in their own modules. In my experience with redux every feature has its own Store, State, Action and Reducer that is derived and communicates with the App Store.

Reply:
I didn't see usage like this it will be interesting to know more